### PR TITLE
Azure Provider Skeleton

### PR DESCRIPTION
**Pull request by logyball**
Adding the skeleton of the Azure Provider.

Note that this doesn't do much, just adds some building blocks and a test to ensure that we can instantiated it.

### CODE REVIEW

File: pkg/azure/azure.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/cloudcost-exporter/pkg/provider"
+)
+
+const (
+	// subsystem = "azure"
+	collectTimeout time.Duration = 5 * time.Minute
```
Comment (Pokom) [inline pkg/azure/azure.go]: I really like this notion, but don't know if this is appropriate at a per provider level. How do you feel about making this an app level configuration that we can then pass to other providers as well?

File: pkg/azure/azure.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	Logger  *slog.Logger
+}
+
+type Config struct {
+	Logger *slog.Logger
+}
+
+// New is a TODO
+func New(ctx context.Context, config *Config) (*Azure, error) {
+	providerGroup := config.Logger.WithGroup("azure")
```
Comment (Pokom) [inline pkg/azure/azure.go:33]: Suggestion: Make `azure` refer to the `subsystem` const that's been commented out

File: pkg/azure/azure.go (hunk 1 above)
Comment (logyball) [reply pkg/azure/azure.go]: Good point, added it as a CLI arg and object on the top-level config

File: pkg/azure/azure.go (hunk 2 above)
Comment (logyball) [reply pkg/azure/azure.go]: done

**Review by Pokom — APPROVED** [review]
Really appreciate chunking this out into a smaller PR. I think I'm going to borrow your `Collect` method creating a context with timeout to pass down to collectors!

**Pull request merged**